### PR TITLE
feat: add donation request component

### LINKED ISSUE #3491: feat: add donation request
After a (logged in) user clicks/taps to download files on a how-to, etc. a modal should be added to invite the user to make a donation.

image

Must haves:
- [x] Generic DonationRequest component in the component library. Responsive. Flexbox. 
- [x] Update Modal to accomidate components that need to be responsive
- [x] Add listener for when iframe changes on donation accepted
- [x] Put behind beta-tester wrapper and only for how-tos
- [x] Test flow on cypress
- [x] Wrap around the DownloadFileFromLink component
- [ ] Wrap around the DownloadStaticFile component
- [ ] All the text controlled by app common component

Should have:
- [x] Common component in the app for the iframe/config option
- [ ] Some config by platform for the iframe
- [ ] Add labels convention to component library

### PR DESCRIPTION
**Pull request by benfurber**
PR Checklist

- [x] - Commit [messages are descriptive](https://github.com/ONEARMY/community-platform/blob/master/CONTRIBUTING.md#--commit-style-guide), it will be used in our [Release Notes](https://github.com/ONEARMY/community-platform/releases/)

PR Type

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Developer experience (improves developer workflows for contributing to the project)

## Description

Adds a component to display a donation request.

With this PR it's only added to external files on howtos for beta-testers.

Part of #3491

### Start of journey
Screenshot 2024-05-15 at 11 50 15

Screenshot 2024-05-15 at 11 49 56
(can scroll)

### End of journey
Screenshot 2024-05-15 at 11 49 10
Screenshot 2024-05-15 at 11 49 26

### CODE REVIEW

**Comment by benfurber** [comment]
I need to have a custom thank you page for the donations. That way, when the iframe url changes, I think I can trigger the download to happen.

**Comment by benfurber** [comment]
For Dalibor. This is the flow. Got any quick ideas on making the thank you page more exciting?

Complete the form:
Screenshot 2024-05-10 at 13 10 15

Then:
Screenshot 2024-05-10 at 13 09 41

**Comment by benfurber** [comment]
@davehakkens Can you send me the image for the modal please?

**Comment by benfurber** [comment]
Need final images, etc. and to tweak the response bits, but basically ready for others to test.

**Comment by davehakkens** [comment]
I dont see a pop-up with the preview-link (i am beta-tester)

**Comment by benfurber** [comment]
@davehakkens Just remembered we always have this problem with the PP only features becuase we don't see the necessary variable for it on the test build. It's _probably_ fine to set that variable for the test build, but it might impact some of the tests. I'll do it on Monday.

**Comment by davehakkens** [comment]
> @davehakkens Just remembered we always have this problem with the PP only features

Ah ok. In this case it don't think it needs to be for PP only because:
1. PP is the only one using how-to module
2. Any other project would want this feature as well if they enable the module (with its own text + image)

**Comment by benfurber** [comment]
@davehakkens Updated. But looking now the responsive elements of the iframe are too messy. I'll shout after I've done more tidying up.

**Comment by davehakkens** [comment]
I see the model now (nice!)
But cant see the donorbox embed. Does it work for you in previewlink?
afbeelding

**Comment by benfurber** [comment]
There's an annoying point where we need to go to vertical alignment but the widget (which I can't control) won't go to 100% width.

Screenshot 2024-05-13 at 11 36 31

I think the padding on left is the best I can do to make it clear it's a 'box in a thing'.

It's now kind of most rubbish at a smaller viewport when it just looks like the padding is off.

Screenshot 2024-05-13 at 11 38 38

**Comment by benfurber** [comment]
@davehakkens Fixed now - I was playing around with urls and forgot to change it back.

**Comment by davehakkens** [comment]
nice. Maybe on mobile make the div same width a the widget (as on this one) 
https://test-pay-what-you-can.webflow.io/home-copy

**Comment by dalibormrska** [comment]
@benfurber - Adding my comments regarding quick improvements of the modal from an UX/UI point of view.

The modal looks overall really nice and functional. Great job!

1. A quick idea would be to add something confetti-like as an image on top of this "Thank you" that pops up after the donation and put more top margin. Right now it sits kind of awkwardly on top.

2. Also differentiating with typography or colour the different information being communicated. The notice about if the download hasn't started yet could be of less visual importance - differentiate it from the thank you message by making it more grey, or stick to the bottom, or for the paragraph to have a background colour...

3. I would also add interpunction - so the exclamation marks - to the "Support our work!" and "Thank you for helping to make this possible!", just so that it looks consistent and more professional (I think it is good to look solid and professional when asking for donation).

4. Is it maybe possible to make the Donorbox colour palette align more with the platform's colour palette? It makes it feel a bit inconsistent with many shades of blue.

image

**Comment by benfurber** [comment]
@davehakkens I've updated the responsive elements. Have another play?

@dalibormrska I like the confetti idea. Could you design me something up? Doesn't have to be in time for this branch - can add later.

**Comment by mariojsnunes** [comment]
for the confetti, this exists https://www.npmjs.com/package/react-confetti :)

**Comment by benfurber** [comment]
@mariojsnunes Ha. Fun. But I was thinking more in keeping with existing graphical style.

**Comment by benfurber** [comment]
@davehakkens I'm using the embed from the login for donorbox that @sigolenej provided, I believe it's being switched over?

**Comment by davehakkens** [comment]
> @davehakkens I'm using the embed from the login for donorbox that @sigolenej provided, I believe it's being switched over?

I dont know which one she send. But indeed its switched over to Precious Plastic with hello@preciousplastic.com login and the "[Community Platform donors](https://donorbox.org/org_admin/forms/581788)" campaign (which looks visually different from yours)

**Comment by dalibormrska** [comment]
Something like this?

image

**Comment by benfurber** [comment]
@davehakkens Got it. Was in the right login but the wrong campaign. I've changed the thank you url now, so until this branch is merged it'll 404. 

@dalibormrska Nice. @davehakkens, have we generally stuck to monocrome graphics for ease or deliberate style?

**Review by mariojsnunes — APPROVED** [review]

File: packages/components/src/DonationRequest/DonationRequest.tsx [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+
+export interface IProps {
+  body: string
+  callback: () => void
+  iframeSrc: string
+  imageURL: string
+  link: string
+}
+
+const REQUEST_TITLE = 'Support our work'
```
[the remark this answers is not in the document]
Comment (benfurber) [reply packages/components/src/DonationRequest/DonationRequest.tsx:15]: Yeah, good question. That currently isn't a convention in the component library but probably should. I'll think about it in a follow-up branch.

**Comment by davehakkens** [comment]
FYI: I still have a few visual remarks but will add them once I can do a full testrun with the Donorbox embed working in a new issue

**Pull request merged**